### ISSUE TITLE
Panic on `local` RemoteStateReference with incorrect file path

### ISSUE THREAD

**Issue by clstokes** [pulumi/pulumi-terraform#497]
When using the _local_ RemoteStateReference implementation, if an incorrect path is used, Pulumi panics.

#### Code
```
import * as terraform from "@pulumi/terraform";
const vpc = new terraform.state.RemoteStateReference("vpc", {
    backendType: "local",
    path: "non-existent-file.json",
});
```

#### Output
```
$ pulumi preview
Previewing update (dev):

     Type                                     Name                              Plan       Info
 +   pulumi:pulumi:Stack                      pulumi-remotestate-reference-dev  create..   	/home/travis/gopath/src/github.com/pulumi/pulum +   pulumi:pulumi:Stack                      pulumi-remotestate-reference-dev  create     1 error; 22 messages
     └─ terraform:state:RemoteStateReference  vpc                                          1 error

Diagnostics:
  terraform:state:RemoteStateReference (vpc):
    error: Preview failed: transport is closing

  pulumi:pulumi:Stack (pulumi-remotestate-reference-dev):
    panic: RootModule called on nil State
    goroutine 51 [running]:
    github.com/pulumi/pulumi-terraform/vendor/github.com/hashicorp/terraform/states.(*State).RootModule(...)
    	/home/travis/gopath/src/github.com/pulumi/pulumi-terraform/vendor/github.com/hashicorp/terraform/states/state.go:93
    github.com/pulumi/pulumi-terraform/pkg/provider.remoteStateReferenceRead(0x2b47f40, 0xc000175da0, 0xc00025a000, 0x24, 0x0, 0x0)
    	/home/travis/gopath/src/github.com/pulumi/pulumi-terraform/pkg/provider/remote_state.go:112 +0x106f
    github.com/pulumi/pulumi-terraform/pkg/provider.(*Provider).Read(0xc000080020, 0x2b47f40, 0xc000175da0, 0xc00025a000, 0xc000080020, 0x24d2a01, 0xc0000486c0)
    	/home/travis/gopath/src/github.com/pulumi/pulumi-terraform/pkg/provider/provider.go:95 +0x8a
    github.com/pulumi/pulumi-terraform/vendor/github.com/pulumi/pulumi/sdk/proto/go._ResourceProvider_Read_Handler.func1(0x2b47f40, 0xc000175da0, 0x267e480, 0xc00025a000, 0x26b4ec0, 0x3ba0130, 0x2b47f40, 0xc000175da0)
    	/home/travis/gopath/src/github.com/pulumi/pulumi-terraform/vendor/github.com/pulumi/pulumi/sdk/proto/go/provider.pb.go:1349 +0x86
    github.com/pulumi/pulumi-terraform/vendor/github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc.OpenTracingServerInterceptor.func1(0x2b47f40, 0xc0001751d0, 0x267e480, 0xc00025a000, 0xc0000aab80, 0xc0000aaba0, 0x0, 0x0, 0x2b0bcc0, 0xc000081930)
    	/home/travis/gopath/src/github.com/pulumi/pulumi-terraform/vendor/github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc/server.go:61 +0x360
    github.com/pulumi/pulumi-terraform/vendor/github.com/pulumi/pulumi/sdk/proto/go._ResourceProvider_Read_Handler(0x2667800, 0xc000080020, 0x2b47f40, 0xc0001751d0, 0xc0000d5f90, 0xc0000aa080, 0x2b47f40, 0xc0001751d0, 0xc0001d8000, 0x9a)
    	/home/travis/gopath/src/github.com/pulumi/pulumi-terraform/vendor/github.com/pulumi/pulumi/sdk/proto/go/provider.pb.go:1351 +0x158
    github.com/pulumi/pulumi-terraform/vendor/google.golang.org/grpc.(*Server).processUnaryRPC(0xc00054a300, 0x2b63d00, 0xc000001680, 0xc0001ca200, 0xc0001742d0, 0x3b6fa08, 0x0, 0x0, 0x0)
    	/home/travis/gopath/src/github.com/pulumi/pulumi-terraform/vendor/google.golang.org/grpc/server.go:971 +0x470
    github.com/pulumi/pulumi-terraform/vendor/google.golang.org/grpc.(*Server).handleStream(0xc00054a300, 0x2b63d00, 0xc000001680, 0xc0001ca200, 0x0)
    	/home/travis/gopath/src/github.com/pulumi/pulumi-terraform/vendor/google.golang.org/grpc/server.go:1250 +0xd25
    github.com/pulumi/pulumi-terraform/vendor/google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc000648040, 0xc00054a300, 0x2b63d00, 0xc000001680, 0xc0001ca200)
    	/home/travis/gopath/src/github.com/pulumi/pulumi-terraform/vendor/google.golang.org/grpc/server.go:690 +0x9f
    created by github.com/pulumi/pulumi-terraform/vendor/google.golang.org/grpc.(*Server).serveStreams.func1
    	/home/travis/gopath/src/github.com/pulumi/pulumi-terraform/vendor/google.golang.org/grpc/server.go:688 +0xa1

    error: preview failed

Permalink: https://app.pulumi.com/clstokes/pulumi-remotestate-reference/dev/previews/b2393f9b-29b0-4003-bcb4-551bba76a3c2
$
```

**Comment by jen20** [comment]
I'll address this once #500 has landed - it should be a straightforward fix.

**Comment by jen20** [comment]
@clstokes Trying to repro this with the latest (v0.18.3) package, it seems it's already been fixed (and the condition is handled here: https://github.com/pulumi/pulumi-terraform/blob/master/pkg/provider/remote_state.go#L113-L115)

`index.ts`:
```typescript
import * as terraform from "@pulumi/terraform";
const vpc = new terraform.state.RemoteStateReference("vpc", {
    backendType: "local",
    path: "non-existent-file.json",
});
```

```shell
$ pulumi up
Previewing update (dev):

     Type                                     Name         Plan       Info
 +   pulumi:pulumi:Stack                      ptf-497-dev  create     1 error
     └─ terraform:state:RemoteStateReference  vpc                     1 error

Diagnostics:
  pulumi:pulumi:Stack (ptf-497-dev):
    error: preview failed

  terraform:state:RemoteStateReference (vpc):
    error: Preview failed: remote state not found
```

Perhaps you have a stray `pulumi-resource-terraform` on your `PATH`?

**Comment by clstokes** [comment]
I tried this in a fresh Vagrant box, confirmed I'm on `0.18.3`, and get the same panic. @danfhernandez gets the same on his machine. Do we need a new release? It looks like stack72's fix was after the 0.18.3 release, no?

```
vagrant@ubuntu-bionic:~/proj$ pulumi up
Enter your passphrase to unlock config/secrets
    (set PULUMI_CONFIG_PASSPHRASE to remember):
Previewing update (dev):

     Type                                     Name      Plan       Info
 +   pulumi:pulumi:Stack                      proj-dev  create.    	/home/travis/gopath/src/github.com/pulumi/pulumi-terraform/vendor/googl +   pulumi:pulumi:Stack                      proj-dev  create     1 error; 22 messages
     └─ terraform:state:RemoteStateReference  vpc                  1 error

Diagnostics:
  terraform:state:RemoteStateReference (vpc):
    error: Preview failed: transport is closing

  pulumi:pulumi:Stack (proj-dev):
    error: preview failed

    panic: RootModule called on nil State
    goroutine 35 [running]:
    github.com/pulumi/pulumi-terraform/vendor/github.com/hashicorp/terraform/states.(*State).RootModule(...)
    	/home/travis/gopath/src/github.com/pulumi/pulumi-terraform/vendor/github.com/hashicorp/terraform/states/state.go:93
    github.com/pulumi/pulumi-terraform/pkg/provider.remoteStateReferenceRead(0x1f51a00, 0xc0004d34a0, 0xc0004a6b40, 0x24, 0x0, 0x0)
    	/home/travis/gopath/src/github.com/pulumi/pulumi-terraform/pkg/provider/remote_state.go:112 +0x106f
    github.com/pulumi/pulumi-terraform/pkg/provider.(*Provider).Read(0xc0003221f0, 0x1f51a00, 0xc0004d34a0, 0xc0004a6b40, 0xc0003221f0, 0x18db301, 0xc000458bc0)
    	/home/travis/gopath/src/github.com/pulumi/pulumi-terraform/pkg/provider/provider.go:95 +0x8a
    github.com/pulumi/pulumi-terraform/vendor/github.com/pulumi/pulumi/sdk/proto/go._ResourceProvider_Read_Handler.func1(0x1f51a00, 0xc0004d34a0, 0x1a87a00, 0xc0004a6b40, 0x1abe200, 0x2fab738, 0x1f51a00, 0xc0004d34a0)
    	/home/travis/gopath/src/github.com/pulumi/pulumi-terraform/vendor/github.com/pulumi/pulumi/sdk/proto/go/provider.pb.go:1349 +0x86
    github.com/pulumi/pulumi-terraform/vendor/github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc.OpenTracingServerInterceptor.func1(0x1f51a00, 0xc0004d3380, 0x1a87a00, 0xc0004a6b40, 0xc000124be0, 0xc000124c00, 0x0, 0x0, 0x1f15420, 0xc0000f9f90)
    	/home/travis/gopath/src/github.com/pulumi/pulumi-terraform/vendor/github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc/server.go:61 +0x360
    github.com/pulumi/pulumi-terraform/vendor/github.com/pulumi/pulumi/sdk/proto/go._ResourceProvider_Read_Handler(0x1a70900, 0xc0003221f0, 0x1f51a00, 0xc0004d3380, 0xc0004a6af0, 0xc0001242e0, 0x1f51a00, 0xc0004d3380, 0xc0000461b0, 0x82)
    	/home/travis/gopath/src/github.com/pulumi/pulumi-terraform/vendor/github.com/pulumi/pulumi/sdk/proto/go/provider.pb.go:1351 +0x158
    github.com/pulumi/pulumi-terraform/vendor/google.golang.org/grpc.(*Server).processUnaryRPC(0xc00012aa80, 0x1f6d4c0, 0xc00012af00, 0xc000504500, 0xc00031e840, 0x2f7abc8, 0x0, 0x0, 0x0)
    	/home/travis/gopath/src/github.com/pulumi/pulumi-terraform/vendor/google.golang.org/grpc/server.go:971 +0x470
    github.com/pulumi/pulumi-terraform/vendor/google.golang.org/grpc.(*Server).handleStream(0xc00012aa80, 0x1f6d4c0, 0xc00012af00, 0xc000504500, 0x0)
    	/home/travis/gopath/src/github.com/pulumi/pulumi-terraform/vendor/google.golang.org/grpc/server.go:1250 +0xd25
    github.com/pulumi/pulumi-terraform/vendor/google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc00032e170, 0xc00012aa80, 0x1f6d4c0, 0xc00012af00, 0xc000504500)
    	/home/travis/gopath/src/github.com/pulumi/pulumi-terraform/vendor/google.golang.org/grpc/server.go:690 +0x9f
    created by github.com/pulumi/pulumi-terraform/vendor/google.golang.org/grpc.(*Server).serveStreams.func1
    	/home/travis/gopath/src/github.com/pulumi/pulumi-terraform/vendor/google.golang.org/grpc/server.go:688 +0xa1

vagrant@ubuntu-bionic:~/proj$ pu
pulumi                         pulumi-language-nodejs         pulumi-resource-pulumi-python
pulumi-analyzer-policy         pulumi-language-python         purge-old-kernels
pulumi-language-go             pulumi-resource-pulumi-nodejs  pushd
```
